controller/authorize.go

```diff
@@ -0,0 +1,89 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/login"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/goadesign/goa"
+	errs "github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+// AuthorizeController implements the authorize resource.
+type AuthorizeController struct {
+	*goa.Controller
+	Auth          login.KeycloakOAuthService
+	TokenManager  token.Manager
+	Configuration LoginConfiguration
+}
+
+func NewAuthorizeController(service *goa.Service, auth *login.KeycloakOAuthProvider, tokenManager token.Manager, configuration LoginConfiguration) *AuthorizeController {
+	return &AuthorizeController{Controller: service.NewController("AuthorizeController"), Auth: auth, TokenManager: tokenManager, Configuration: configuration}
+}
+
+// Authorize runs the authorize action.
+func (c *AuthorizeController) Authorize(ctx *app.AuthorizeAuthorizeContext) error {
+
+	if ctx.State == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("state", "nil").Expected("State"))
+	}
+
+	if ctx.Code == nil {
+		if ctx.ClientID == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("client_id", "nil").Expected("Service Account ID"))
+		}
+		if ctx.RedirectURI == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("redirect_uri", "nil").Expected("Redirect URI"))
+		}
+		if ctx.ResponseType == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("response_type", "nil").Expected("Response Type"))
+		}
+		if ctx.Scope == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("scope", "nil").Expected("Scope"))
+		}
+
+		_, found := c.Configuration.GetServiceAccounts()[*ctx.ClientID]
+		if !found {
+			log.Error(ctx, map[string]interface{}{
+				"client_id": *ctx.ClientID,
+			}, "Unknown Service Account ID")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("invalid Service Account ID"))
+		}
+
+	}
+	authEndpoint, err := c.Configuration.GetKeycloakEndpointAuth(ctx.RequestData)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "Unable to get Keycloak Auth endpoint URL")
```

Comment:
error messages in small letters ( we try to follow this convention )

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/210/commits/391604b2affa2b852131a55a5f94a7d1b120c10c